### ISSUE TITLE
Hovering the cursor over 'var' when debugging

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#28602]
In a C# application, it is not always obvious understanding the type represented from the 'var' declarator.
For example:
var result = OneMethod(...);

At development time, hovering the mouse over 'var' works and you can see the effective type that will be assigned to the variable.

When debugging, hovering over 'var' does not produce any result, hence you can't know what is the type of the variable.

Please, add the 'var' hovering support at debugging time too.

_This issue has been moved from https://developercommunity.visualstudio.com/content/problem/294777/hovering-the-cursor-over-var-when-debugging.html
VSTS ticketId: 649484_
_These are the original issue comments:_
(no comments)
_These are the original issue solutions:_
(no solutions)

**Comment by Neme12** [comment]
I agree. It can be annoying that IDE features are not available during debug and especially edit-and-continue development.

**Comment by Neme12** [comment]
There are challenges of course. What if there's a debug tooltip at the same location? Do we hide the IDE one and only make it available under a keybord shortcut? (probably not discoverable - such a feature may already exist and I wouldn't know about it). Do we somehow show both? (I can see how this would be intrusive during debugging)

![image](https://user-images.githubusercontent.com/11444821/42818598-92fe2d7e-89d1-11e8-89a5-023c67c8203c.png)

**Comment by sharwell** [comment]
@Neme12 This issue related to the `var` token, not to the `x` token in your example, so there shouldn't be a conflict.

**Comment by Neme12** [comment]
So the path forward will be to choose a closed set of contexts/tokens where quick info will be enabled?

**Comment by MarkusAmshove** [comment]
I would propose to also use CodeLens to show infered types at *debug* time.

In IntelliJ with Kotlin you can enable `Type hints` which show the infered type:

![grafik](https://user-images.githubusercontent.com/2401875/42926473-cbcc1902-8b31-11e8-9c7b-cd0ae39de591.png)

**Comment by dpoeschl** [comment]
For this specific case, don't pass the `var` identifier to the EE when it's the contextual keyword, and instead show QuickInfo. More generally, we'd like to show the datatip if one is available, or regular Quick Info if one is not available.

(Getting _both_ the datatip and quick info at the same time would require Editor work.)
